feat(kraft): Introduce the volume subcommand

LucaSeri opened the pull request:
<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

  Kraftkit follows the same guidelines as the Unikraft Open Source Project.

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [x] Updated relevant documentation.

### Description of changes

This ``PR`` introduces the ``volume`` subcommand and also allows users to run machines with named volumes (as opposed to paths on host).

<!--
Please provide a detailed description of the changes made in this new PR.
-->

nderjung commented on internal/cli/kraft/run/utils.go:
```diff
@@ -201,7 +201,7 @@ func (opts *RunOptions) parseVolumes(ctx context.Context, machine *machineapi.Ma
 			mountPath = split[1]
 		} else {
-			return fmt.Errorf("invalid syntax for --volume=%s expected --volume=<host>:<machine>", volLine)
+			return fmt.Errorf("invalid syntax for --volume=%s expected --volume=<host>/<vol_name>:<machine>", volLine)
```
Im not sure what this additional syntax implies

Commit:
feat(api): Add new Managed flag to VolumeSpec

The Manged flag differentiates named volumes whose
storage (creation, deletion) is done by kraftkit itself from
volumes which simply refer to a path on the host, which the
user is managing.

Signed-off-by: Luca Seritan <luca.seritan@gmail.com>

Commit:
feat(9pfs): Allow named volumes

Named volumes are volumes managed by kraftkit. They get their
own path in the RuntimeDir whose lifetime is managed by kraft.

Signed-off-by: Luca Seritan <luca.seritan@gmail.com>

Commit:
feat(volume): Add volume service iterator

The volume service iterator is useful for performing
operations on volumes when the underlying driver is not
known.

Signed-off-by: Luca Seritan <luca.seritan@gmail.com>

Commit:
feat(run): Support named volumes

Whenever a volume argument host:machine is passed, the host
parameter is first checked against the volume store. In case
the name matches that of a named volume, that volume is used.

Signed-off-by: Luca Seritan <luca.seritan@gmail.com>

Commit:
feat(start): Update volume state

Signed-off-by: Luca Seritan <luca.seritan@gmail.com>

Commit:
feat(remove): Update volume status

Signed-off-by: Luca Seritan <luca.seritan@gmail.com>

Commit:
feat(volume): Add the create subcommand

The create subcommand creates a named volume.

Signed-off-by: Luca Seritan <luca.seritan@gmail.com>

Commit:
feat(volume): Add inspect subcommand

The inspect subcommand allows printing information about
a volume.

Signed-off-by: Luca Seritan <luca.seritan@gmail.com>

Commit:
feat(volume): Add list subcommand

Signed-off-by: Luca Seritan <luca.seritan@gmail.com>

Commit:
feat(volume): Add remove subcommand

Signed-off-by: Luca Seritan <luca.seritan@gmail.com>

nderjung commented on api/volume/v1alpha1/volume.zip.go:
```diff
@@ -39,6 +39,10 @@ type VolumeSpec struct {
+
+	// ManagedInternally is a flag that indicates whether the volume is managed
+	// by kraftkit or not.
+	ManagedInternally bool `json:"managedInternally,omitempty"`
```
Let's just call this `Managed`.

nderjung reviewed (approved):
Thanks!
